rides/__init__.py

```diff
@@ -384,6 +402,77 @@ def leave_ride(car_id, rider_username):
     if rider.username == username and rider is not None:
         db.session.delete(rider)
         car.current_capacity -= 1
+        notify_opening( car.event_id, car.name, car_id )
         db.session.add(car)
         db.session.commit()
     return redirect(url_for('index'))
+
+# Automatically leave a ride and join different ride, used for Needs a Ride Notifications
+@app.route('/autojoin/<string:leave_id>/<string:join_id>/<string:user>', methods=['GET'])
+@login_required
+def autojoin(leave_id, join_id, user):
+    # LEAVE
+    username = current_user.id
+    car = Car.query.get(leave_id)
+    rider = Rider.query.filter(Rider.username == user, Rider.car_id == leave_id).first()
+    if rider.username == username and rider is not None:
+        db.session.delete(rider)
+        car.current_capacity -= 1
+        db.session.add(car)
+        db.session.commit()
+    # JOIN
+    incar = False
+    #username = current_user.id
+    name = current_user.firstname + " " + current_user.lastname
+    slack = current_user.slack
+    email = current_user.email
+    car = Car.query.get(join_id)
+    event = Event.query.get(car.event_id)
+    attempted_username = user
+    if attempted_username == username:
+        for c in event.cars:
+            if c.username == username:
+                incar = True
+            for person in c.riders:
+                if person.username == username:
+                    incar = True
```

Comment:
This seems inefficient... Are we sure this can't be done in one fell query to the DB

Reply:
I think I might know a way, will look into it

Reply:
Changed in commit https://github.com/ComputerScienceHouse/csh-rideboard/pull/22/commits/711d1a4ad688e98faa8a7ae76c4c812854517b00